packages/flutter/test/widgets/image_test.dart

```diff
@@ -1956,6 +1957,18 @@ void main() {
       matchesGoldenFile('image_test.missing.2.png'),
     );
   }, skip: kIsWeb); // https://github.com/flutter/flutter/issues/74935 (broken assets not being reported on web)
+
+   testWidgets('Image.file throws a non-implemented error on web', (WidgetTester tester) async {
+    const String expectedError =
+      'Image.file is not supported on Flutter Web. '
+      'Consider using either Image.asset or Image.network instead.';
+    final Uri uri = Uri.parse('/home/flutter/dash.png');
+    final File file = File.fromUri(uri);
+    expect(
+      () => Image.file(file),
+      throwsA(predicate((AssertionError e) => e.message == expectedError)),
+    );
+  }, skip: !kIsWeb); // https://github.com/flutter/flutter/issues/88772 (Improve the error message when calling Image.file on web)
```

Comment:
Actually, I think I was wrong. Using `skip` this way goes against the spirit of the skip policy. The linked issue is meant to remind us to remove the `skip` in the future, but in this case the skip is intentional, we want it to stay forever. So instead, I propose a different approach: remove `skip` and instead test for different behaviors on different platforms as follows:

```dart
expect(
  () => Image.File(file),
  kIsWeb
    // Web does not support file access, expect AssertionError
    ? throwsA(predicate((AssertionError e) => e.message == expectedError))
    // AOT supports file access, expect constructor to succeed
    : isNot(throws),
),
```

Reply:
Done. PTAL